src/app/clients/add/submitFormHelpers.ts

```diff
@@ -0,0 +1,318 @@
+    const childrenInDatabase = await getChildrenInDatabase(familyID);
+    const people: PeopleGroupedByAction = { toInsert: [], toUpdate: [], toDelete: [] };
+
+    for (const child of children) {
+        if (child.primaryKey === undefined) {
+            people.toInsert.push(child);
+            continue;
+        }
+        if (childrenInDatabase.includes(child.primaryKey)) {
+            people.toUpdate.push(child);
+            continue;
+        }
+        people.toDelete.push(child);
+    }
+
+    for (const adult of adults) {
+        const numberAdultsInDatabase = await getNumberAdults(familyID, adult.gender);
+        const difference = adult.quantity! - numberAdultsInDatabase;
+        if (difference > 0) {
+            const record = { ...adult, quantity: difference };
+            people.toInsert.push(record);
+        }
+        if (difference < 0) {
+            await deleteAdultMembers(familyID, adult.gender, difference);
+        }
+    }
+
+    await insertFamily(people.toInsert, familyID);
+    await updateChildren(people.toUpdate);
+    await deleteChildren(people.toDelete);
+};
+
+const revertClientInsert = async (primaryKey: string): Promise<void> => {
+    await supabase.from("clients").delete().eq("primary_key", primaryKey);
+};
+
+// TODO VFB-22: Handle error from Family Updates
+const revertClientUpdate = async (initialRecords: ClientDatabaseUpdateRecord): Promise<void> => {
+    await supabase
+        .from("clients")
+        .update(initialRecords)
+        .eq("primary_key", initialRecords.primary_key);
+};
+
+const formatClientInsertRecord = (fields: Fields): ClientDatabaseInsertRecord => {
+    const extraInformationWithNappy =
+        fields.nappySize === ""
+            ? fields.extraInformation
+            : `Nappy Size: ${fields.nappySize}, Extra Information: ${fields.extraInformation}`;
+
+    return {
+        full_name: fields.fullName,
+        phone_number: fields.phoneNumber,
+        address_1: fields.addressLine1,
+        address_2: fields.addressLine2,
+        address_town: fields.addressTown,
+        address_county: fields.addressCounty,
+        address_postcode: fields.addressPostcode,
+        dietary_requirements: checkboxGroupToArray(fields.dietaryRequirements),
+        feminine_products: checkboxGroupToArray(fields.feminineProducts),
+        baby_food: fields.babyProducts,
+        pet_food: checkboxGroupToArray(fields.petFood),
+        other_items: checkboxGroupToArray(fields.otherItems),
+        delivery_instructions: fields.deliveryInstructions,
+        extra_information: extraInformationWithNappy,
+        signposting_call_required: fields.signpostingCall,
+        flagged_for_attention: fields.attentionFlag,
+    };
+};
+
+const formatClientUpdateRecord = (fields: Fields): ClientDatabaseUpdateRecord => {
+    // TODO VFB-24: Not sure if I should just make this and formatClientInsertRecord 1 function because they are the same, but are technically different type.
```

Comment:
looks copy-pasted, I'd go with just using the one (formatClientRecord?)
maybe have a quick look into if you can give it a return type of `Insert | Update` - though might be more effort than is worth!